### PR TITLE
feat: updates code feature block ui

### PR DESCRIPTION
**Pull request by paulpopus**
CMS PR: https://github.com/payloadcms/website-cms/pull/58

Old:
image
image

New:
image
image

### CODE REVIEW

**Comment by PatrikKozak** [comment]
@paulpopus The `code` portion of this block is still changing `height` size when the tabs have different lengths of code - didn't we decide we don't want the height changing?

File: src/blocks/CodeFeature/index.tsx [hunk 1]
```diff
@@ -1,78 +1,145 @@
...
+              className={[classes.scanlineDesktop].filter(Boolean).join(' ')}
+              crosshairs={['top-left', 'bottom-left']}
+            />
+            {/* <CrosshairIcon className={[classes.crosshairTopLeft].filter(Boolean).join(' ')} /> */}
+            <CrosshairIcon className={[classes.crosshairTopRight].filter(Boolean).join(' ')} />
+            {/* <CrosshairIcon className={[classes.crosshairBottomLeft].filter(Boolean).join(' ')} /> */}
+            <CrosshairIcon className={[classes.crosshairBottomRight].filter(Boolean).join(' ')} />
+          </div>
+          <div className={[classes.content, 'cols-4 cols-m-8'].filter(Boolean).join(' ')}>
+            <h2 className={classes.heading}>{heading}</h2>
```
Comment (PatrikKozak) [inline src/blocks/CodeFeature/index.tsx:64]: @paulpopus I am seeing this as an `h6` in the design - am I looking at an old design or should this be an `h6` instead?

File: src/blocks/CodeFeature/index.tsx [hunk 2]
```diff
@@ -1,78 +1,145 @@
...
+        <div className={[classes.container, 'grid'].filter(Boolean).join(' ')}>
+          <BackgroundGrid ignoreGutter className={classes.backgroundGrid} />
+          <div className={[classes.scanlineWrapper, 'start-9 cols-8'].filter(Boolean).join(' ')}>
+            <BackgroundScanline
+              className={[classes.scanlineDesktop].filter(Boolean).join(' ')}
+              crosshairs={['top-left', 'bottom-left']}
+            />
+            {/* <CrosshairIcon className={[classes.crosshairTopLeft].filter(Boolean).join(' ')} /> */}
+            <CrosshairIcon className={[classes.crosshairTopRight].filter(Boolean).join(' ')} />
+            {/* <CrosshairIcon className={[classes.crosshairBottomLeft].filter(Boolean).join(' ')} /> */}
```
Comment (PatrikKozak) [inline src/blocks/CodeFeature/index.tsx:60]: @paulpopus remove commented out code

**Comment by PatrikKozak** [comment]
@paulpopus When only one tab in this block - this extra border on top of the code block should not be there

Screen Shot 2024-02-02 at 1 07 03 PM

**Comment by PatrikKozak** [comment]
@paulpopus Is this small white box in the bottom right corner intended? Looks odd to me

Screen Shot 2024-02-02 at 1 06 27 PM

File: src/blocks/CodeFeature/index.tsx [hunk 3]
```diff
@@ -1,78 +1,145 @@
...
+            <CrosshairIcon className={[classes.crosshairBottomRight].filter(Boolean).join(' ')} />
+          </div>
+          <div className={[classes.content, 'cols-4 cols-m-8'].filter(Boolean).join(' ')}>
+            <h2 className={classes.heading}>{heading}</h2>
+            {forceDarkBackground}
```
Comment (PatrikKozak) [inline src/blocks/CodeFeature/index.tsx:65]: @paulpopus Why is this `{forceDarkBackground}` here?

File: src/blocks/CodeFeature/index.tsx [hunk 4]
```diff
@@ -1,78 +1,145 @@
...
+            <div className={[''].filter(Boolean).join(' ')}>
+              <div className={''}>
```
Comment (PatrikKozak) [inline src/blocks/CodeFeature/index.tsx:67]: @paulpopus remove `className` from these if there are no classes being added to the elements & are these wrapping elements even needed then?

**Comment by PatrikKozak** [comment]
@paulpopus seeing this error below in `index.tsx` of `CodeFeature`:

Screen Shot 2024-02-02 at 1 18 56 PM

Are the types not updated?

File: src/blocks/CodeFeature/index.tsx (hunk 1 above)
Comment (paulpopus) [reply src/blocks/CodeFeature/index.tsx:64]: Yeah it's h2 here 
image

**Comment by paulpopus** [comment]
> @paulpopus The `code` portion of this block is still changing `height` size when the tabs have different lengths of code - didn't we decide we don't want the height changing?

I understood it that we just want a min height so it doesn't jump between the smaller codeblocks and then _some_ max height so it scrolls if it's really too big

**Comment by paulpopus** [comment]
> Are the types not updated?

@PatrikKozak They are, I just checked the payload-types.ts file, maybe you need to reload the types when changing to this branch?

File: src/blocks/CodeFeature/index.module.scss [hunk 5]
```diff
@@ -1,16 +1,31 @@
...
+      z-index: 0;
+    }
+  }
 }
 
-.codeFeatureGrid {
+.container {
+  position: relative;
   align-items: center;
+  row-gap: calc(var(--base) * 3);
```
Comment (PatrikKozak) [inline src/blocks/CodeFeature/index.module.scss:16]: @paulpopus I would reduce the `row gap` in `mid` and `small` breaks. Seems like too much spacing in mobile

Screen Shot 2024-02-05 at 11 55 04 AM

**Pull request merged**